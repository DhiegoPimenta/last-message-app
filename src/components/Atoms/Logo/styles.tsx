import styled, { css } from 'styled-components'
import { LogoProps } from '.'

const wrapperModifiers = {
    small: () => css`
        width: 12rem;
        height: 12rem;
    `,
    normal: () => css`
        width: 15rem;
        height: 15rem;
    `,
    large: () => css`
        width: 22.6rem;
        height: 22.6rem;
    `,
    center: () => css`
        margin: 2rem auto;
    `,
}

export const Logo = styled.img<LogoProps>`
    ${({ size, center }) => css`
        img {
        }
        width: 100%;

        ${!!size && wrapperModifiers[size!]()}
        ${!!center && wrapperModifiers.center()}
    `}
`
