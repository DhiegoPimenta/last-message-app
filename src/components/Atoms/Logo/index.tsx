import * as S from './styles'
import Link from 'next/link'
import LinkTo from '../Link'

export interface LogoDefaultProps {
    navigateToHome?: boolean
    center?: boolean
}

export interface LogoProps extends LogoDefaultProps {
    size?: 'small' | 'normal' | 'large'
}

const LogoWrapper: React.FC<LogoDefaultProps> = ({
    navigateToHome,
    children,
}) => {
    if (navigateToHome) {
        return (
            <Link href="/">
                <a data-testid="logo-link">{children}</a>
            </Link>
        )
    } else {
        return <>{children}</>
    }
}

const Logo = ({ navigateToHome, size = 'normal', center }: LogoProps) => (
    <LinkTo url="/inicio">
        <LogoWrapper navigateToHome={navigateToHome}>
            <S.Logo
                src="/img/logo.png"
                alt="Logo Last Message"
                size={size}
                center={center}
            />
        </LogoWrapper>
    </LinkTo>
)

export default Logo
