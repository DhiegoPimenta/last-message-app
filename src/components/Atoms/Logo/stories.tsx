import { Story, Meta } from '@storybook/react/types-6-0'

import Logo, { LogoProps } from '.'

export default {
    title: 'Atoms/Logo',
    component: Logo,
} as Meta

export const Basic: Story<LogoProps> = (args) => <Logo {...args} />
Basic.args = {
    size: 'normal',
    navigateToHome: true,
}
