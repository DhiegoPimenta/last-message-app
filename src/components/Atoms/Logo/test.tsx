import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Logo from '.'

describe('<Logo />', () => {
    it('should render a white label by default', () => {
        const { container } = renderWithTheme(<Logo />)

        expect(screen.getByAltText(/logo last message/i)).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a bigger logo', () => {
        renderWithTheme(<Logo size="large" />)

        expect(screen.getByAltText(/logo last message/i)).toHaveStyle({
            width: '22.6rem',
            height: '22.6rem',
        })
    })

    it('should render a smaller logo', () => {
        renderWithTheme(<Logo size="small" />)

        expect(screen.getByAltText(/logo last message/i)).toHaveStyle({
            width: '12rem',
            height: '12rem',
        })
    })

    it('should render a mediun logo by default', () => {
        renderWithTheme(<Logo />)

        expect(screen.getByAltText(/logo last message/i)).toHaveStyle({
            width: '15rem',
            height: '15rem',
        })
    })

    it('should render a link when prop "navigateToHome" is true', () => {
        const { container } = renderWithTheme(<Logo navigateToHome />)

        const link = screen.getByTestId('logo-link')

        expect(link).toBeInTheDocument()
        expect(link.getAttribute('href')).toBe('/')

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should not render a link when prop "navigateToHome" is false', () => {
        renderWithTheme(<Logo />)

        expect(() => screen.getByTestId('logo-link')).toThrowError()
    })
})
