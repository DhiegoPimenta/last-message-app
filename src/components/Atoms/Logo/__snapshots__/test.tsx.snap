// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render a link when prop "navigateToHome" is true 1`] = `
.c0 {
  width: 100%;
  width: 15rem;
  height: 15rem;
}

<a
  data-testid="logo-link"
  href="/"
>
  <img
    alt="Logo Last Message"
    class="c0"
    src="/img/logo.png"
  />
</a>
`;

exports[`<Logo /> should render a white label by default 1`] = `
.c0 {
  width: 100%;
  width: 15rem;
  height: 15rem;
}

<img
  alt="Logo Last Message"
  class="c0"
  src="/img/logo.png"
/>
`;
